zerver/lib/digest.py

```diff
@@ -350,24 +368,31 @@ def bulk_get_digest_context(
     user_stream_map = get_user_stream_map(user_ids, cutoff_date)
 
     for user in users:
-        stream_ids = user_stream_map[user.id]
-
-        recent_topics = []
-        for stream_id in stream_ids:
-            recent_topics += get_recent_topics(realm.id, stream_id, cutoff_date)
-
-        hot_topics = get_hot_topics(recent_topics, stream_ids)
-
         context = common_context(user)
 
         # Start building email template data.
         unsubscribe_link = one_click_unsubscribe_link(user, "digest")
         context.update(unsubscribe_link=unsubscribe_link)
 
-        # Get context data for hot conversations.
-        context["hot_conversations"] = [
-            hot_topic.teaser_data(user, stream_id_map) for hot_topic in hot_topics
-        ]
+        if not message_content_allowed_in_missedmessage_emails(user):
+            # Count new messages when message content is hidden in email notifications.
+            context["new_messages_count"] = get_new_messages_count(user, cutoff_date)
+            context["hot_conversations"] = []
+        else:
+            context["new_messages_count"] = 0
+
+            stream_ids = user_stream_map[user.id]
+
+            recent_topics = []
+            for stream_id in stream_ids:
+                recent_topics += get_recent_topics(realm.id, stream_id, cutoff_date)
+
+            hot_topics = get_hot_topics(recent_topics, stream_ids)
+
+            # Get context data for hot conversations.
```

Comment:
I'd move this up to be just after the `else` line...

```python
else:
  # Otherwise, get context data for hot conversations.
```

And set the new message count to zero last.